src/utils.jl

```diff
@@ -1,14 +1,98 @@
 ## macro for argument checking
 
-macro check_args(D, cond)
-    quote
-        if !($(esc(cond)))
-            throw(ArgumentError(string(
-                $(string(D)), ": the condition ", $(string(cond)), " is not satisfied.")))
+macro strip_linenos(expr)
+    return esc(Base.remove_linenums!(expr))
+end
+
+"""
+    @check_args(
+        D,
+        @setup(statements...),
+        (arg₁, cond₁, message₁),
+        (cond₂, message₂),
+        ...,
+    )
+
+A convenience macro that generates AD-compatible checks of arguments for a distribution of
+type `D`.
+
+More concretely, it generates the following Julia code:
+```julia
+Distributions.check_args(check_args) do
```

Comment:
Perhaps document that it expects a boolean variable `check_args` to be defined?

Reply:
Alternatively, one could allow users/developers to provide the name of the boolean variable. I thought about it since it's more obvious - but it seemed unnecessarily verbose since we only use `check_args`.

Reply:
Yeah, it's only internal, so I don't see the benefit of making it configurable. And this way it actually enforces standardizing the keyword argument.

Reply:
I updated the docstring.